actionpack/lib/action_dispatch/middleware/flash.rb

```diff
@@ -45,6 +45,7 @@ module RequestMethods
       # read a notice you put there or <tt>flash["notice"] = "hello"</tt>
       # to put a new one.
       def flash
+        return Flash::NullFlash unless session.respond_to?(:loaded?)
```

Comment:
I'm not too fan of that `respond_to?` What is `session` returning when disabled? I'd rather call an actual method. e.g `unless session.disabled?`

Ohter than that the PR looks good to me now.

Reply:
Agree. It returns a hash (empty in the text example, but I guess it could be populated too), which is the default Rack session. I noticed this pattern is used in the following places:

https://github.com/rails/rails/blob/8efcf9b/actionpack/lib/action_dispatch/middleware/flash.rb#L70
https://github.com/rails/rails/blob/8efcf9b/actionpack/lib/action_dispatch/http/request.rb#L365

I can change it to something like this if that's better:

```suggestion
        return Flash::NullFlash if Hash === session
```

Reply:
Hum, no I think checking for Hash type is even worse, as it's even less clear what we are checking for.

I'll have a deeper look tomorrow, see if I can find something better. Otherwise I see we already have a bunch of similar `respond_to?` checks, so I don't think we'd be making anything worse.